compiler/src/dotty/tools/dotc/interactive/InteractiveDriver.scala

```diff
@@ -182,14 +182,24 @@ class InteractiveDriver(settings: List[String]) extends Driver {
    *  of a previous run. Note that typed trees can have untyped or partially
    *  typed children if the source contains errors.
    */
-  private def cleanup(tree: tpd.Tree)(implicit ctx: Context): Unit = tree.foreachSubTree { t =>
-    if (t.hasType) {
-      if (t.symbol.exists) {
-        if (!t.symbol.isCompleted) t.symbol.info = UnspecifiedErrorType
-        t.symbol.annotations.foreach(annot => cleanup(annot.tree))
+  private def cleanup(tree: tpd.Tree)(implicit ctx: Context): Unit = {
+    val cleanedTree = mutable.Set.empty[tpd.Tree]
```

Comment:
I think I would call this variable `seen`, this is the name we commonly use in the compiler for loop detection.